src/battle_util.c

```diff
@@ -3894,18 +3894,20 @@ static const u16 sWeatherFlagsInfo[][3] =
     [ENUM_WEATHER_FOG] = {B_WEATHER_FOG_TEMPORARY, B_WEATHER_FOG_PERMANENT, HOLD_EFFECT_NONE},
 };
 
-static void ShouldChangeFormInWeather(u32 battler)
+static void ShouldChangeFormInWeather()
 {
     int i;
-    int side = GetBattlerSide(battler);
-    struct Pokemon *party = GetSideParty(side);
+    int side;
+    struct Pokemon *party; 
 
-    for (i = 0; i < PARTY_SIZE; i++)
+    for (i = 0; i < gBattlersCount; i++)
```

Comment:
why was this using party size previously?

Reply:
It was looping through the party to set the flag for Ice Face to be able to trigger again, maybe a failed attempt at making Ice Face work on switch in.

Reply:
oh, I see